core/src/main/java/com/datastax/oss/driver/internal/core/metadata/SniEndPoint.java

```diff
@@ -64,7 +64,7 @@ public InetSocketAddress resolve() {
       // The order of the returned address is unspecified. Sort by IP to make sure we get a true
       // round-robin
       Arrays.sort(aRecords, IP_COMPARATOR);
-      int index = (aRecords.length == 1) ? 0 : (int) OFFSET.getAndIncrement() % aRecords.length;
+      int index = (aRecords.length == 1) ? 0 : Math.abs(OFFSET.getAndIncrement()) % aRecords.length;
```

Comment:
I think `Math.abs` will yield `Integer.MIN_VALUE` for `Integer.MAX_VALUE + 1`.

This looks safer to me:

```suggestion
      int index = (aRecords.length == 1) ? 0 : OFFSET.getAndUpdate(x -> x == Integer.MAX_VALUE ? 0 : x + 1) % aRecords.length;
```